### ISSUE TITLE
Make conflict resolution strategies configurable

### ISSUE THREAD

**Issue by snikolayev** [NRules/NRules#63]
Currently, conflict resolution strategy in NRules is fixed. Rules with higher priorities are fired first. Rules with the same priority are fired in a FIFO order.
Need ability to control conflict resolution at a more granular level.

**Comment by dynamicscrmbiz** [comment]
Agree. Also, is it possible to add an attribute which allows us to group rules and then the priority configuration can work over a group of rules to decide whether all rules should be fired or certain rules within that group
